packages/jest-mock/src/index.ts

```diff
@@ -1408,13 +1439,14 @@ export class ModuleMocker {
   }
 
   resetAllMocks(): void {
-    this._spyState.forEach(reset => reset());
-    this._mockConfigRegistry = new WeakMap();
     this._mockState = new WeakMap();
+    this._mockConfigRegistry = new WeakMap();
+    this._spyState.forEach(spyState => spyState.reset?.());
   }
 
   restoreAllMocks(): void {
-    this._spyState.forEach(restore => restore());
+    this._mockState = new WeakMap();
```

Comment:
Found this while writing tests. Currently `jest.restoreAllMock()` does not clear mock calls count (and friends), but `mockfn.mockRestore()` calls `mockFn.mockReset` which does call `.mockClear()` so in the end mock calls count is cleared:

https://github.com/facebook/jest/blob/8313d003afe3abb35f247ae72eb1968cacf8a7b5/packages/jest-mock/src/index.ts#L789-L792

@SimenB How do you think, should restoring the original implementation clear mock calls count (and friends) or not? I think that is fine to do. Better to split this to separate PR, right?

Reply:
Yeah, I think clearing makes sense